### PR TITLE
Integrate socket.io for countdown clock

### PR DESCRIPTION
**Pull request by huyanh7796**

### CODE REVIEW

**Review by nddmanh — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "react-router-dom": "5.2.0",
     "react-router-guards": "1.0.0",
     "react-scripts": "4.0.3",
+    "socket.io-client": "^4.1.3",
```
Comment (lethemanh) [inline package.json]: Use fixed version

File: src/modules/playRoom/components/CountDown.js [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
-      <p className="countdown-time">
-        Thời gian còn lại để đặt cược: 
-        <br/>
-        <strong> 59 s </strong>
-      </p>
+      {
+        isProcessingResult ? 
+        <p className="countdown-time">
+          Đang xử lý kết quả...
+        </p> :
```
Comment (lethemanh) [inline src/modules/playRoom/components/CountDown.js]: why ```:``` is placed in this line?

File: src/modules/playRoom/components/CountDown.js [hunk 3]
```diff
@@ -1,11 +1,35 @@
+import { useState, useEffect } from 'react';
+import { socket } from '../../../core/services/socket';
+
 const CountDown = () => {
+  const [count, setCount] = useState(60);
```
Comment (lethemanh) [inline src/modules/playRoom/components/CountDown.js]: Please use const for numbers of second

File: src/index.js [hunk 4]
```diff
@@ -2,6 +2,9 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './App';
 import './assests/style/App.css';
+import { initConnection } from './core/services/socket';
```
Comment (lethemanh) [inline src/index.js:4]: Pull js module to upper

File: src/modules/playRoom/components/CountDown.js (hunk 2 above)
Comment (huyanh7796) [reply src/modules/playRoom/components/CountDown.js]: it's conditional rendering syntax

Comment (lethemanh) [reply src/modules/playRoom/components/CountDown.js]: Place it in front of false condition, it help your code more clearly

Comment (huyanh7796) [reply src/modules/playRoom/components/CountDown.js]: In my opinion, the way I wrote is more clearly. It separates 2 conditions and it's the way Shit taught.

Comment (lethemanh) [reply src/modules/playRoom/components/CountDown.js]: Well I don't write this code so when I read this code I though ```:``` will be render to UI but not part of condition expression

File: package.json [hunk 5]
```diff
@@ -18,6 +18,7 @@
     "react-router-dom": "5.2.0",
     "react-router-guards": "1.0.0",
     "react-scripts": "4.0.3",
+    "socket.io-client": "4.1.3",
```
Comment (lethemanh) [inline package.json:21]: Did you check version of ```socket.io-client``` in package-lock.json?

File: src/modules/playRoom/components/CountDown.js (hunk 2 above)
Comment (huyanh7796) [reply src/modules/playRoom/components/CountDown.js]: it's okay. Im gonna rewrite those codes follow instruction of react doc

File: package.json (hunk 5 above)
Comment (huyanh7796) [reply package.json:21]: In my package-lock.json: "socket.io-client": "^4.1.3"

**Review by lethemanh — APPROVED** [review]

**Pull request merged**